model_evaluation/rfc.md

```diff
@@ -0,0 +1,91 @@
+|     model_uri              |   String        |   Required: False (Required only if mode is score or predict)  Description: MLFlow model uri (could be of type) ->        To Fetch models from an azureml run        - runs:/<azureml_run_id>/run-relative/path/to/model        To Fetch model from azureml model registry        - models:/<model_name>/<model_version>        - models:/<model_name>/<stage>  NOTE: model_uri has been added to accommodate current ML Designer as it doesn’t allow users to drag-and-drop Registered models    |
+|     mlflow_model           |   MLFlow Model  |   Required: False (Required only if mode is score or predict)  Description: MLFlow model (either registered or output of another job)                                                                                                                                                                                                                                                                                                                                                             |
+|     test_data              |   MLTable       |   Required: True  Description: Test Dataset for model evaluation. Allowed formats: Delimited, JSON Lines, Parquet and Plain Text (for some tasks).                                                                                                                                                                                                                        |
+|     evaluation_config      |   MLTable       |   Required: False  Description: Additional config files required by model/metrics package. This data asset should contain a JSON Config file and any number artifacts required to load/run the model. The MLTable file should only mention JSON file name in path.        Contents of JSON File:        - See Additional Parameters list below                               |
+
+#### Additional Parameters :
+
+1. Mode Dependent:
+  - `label_column_name` : String   
+    Target column name in test data. Required for mode = score/compute_metrics
+  - `prediction_column_name` : String   
+    Predictions column name in test data. Required for mode = compute_metrics
+  
+2. Task Dependent:
+  - Classification  
+    a. `metrics` : Optional[List[String]]    
+    List of metric names to be computed. If not provided we choose the default set.  
+    b. `class_labels`: Optional[List[Any]]  
+    List of labels for entire data (all the data train/test/validation)  
+    c. `train_labels:` : Optional[List[Any]]  
+    List of labels used during training.  
+    d. `sample_weight` : Optional[List[Float]]   
+    Weights for the samples (Does not need to match sample weights on the fitted model)  
+    e. `y_transformer` : Optional[String]   
+    Name of transformer (label transformer) file in data_folder. (Only sklearn based transformers are supported for now)  
+    f. `use_binary`: Optional[bool]   
```

Comment:
Can we auto-detect this from the model?

Reply:
or from length of `train_labels` ?

Reply:
Yes, these are currently inferred directly from train_labels. All the task dependent additional parameters are optional. User can provide them if he wants else all of these can be inferred from test data/model.